### DISCUSSION TITLE
Keyboard shortcut overlay

### DISCUSSION THREAD

**Discussion by bakkeby** [bakkeby/dwm-flexipatch#144] [Ideas]
One idea I had two years ago and never pursued is having a kind of overlay where you can have a quick look at your keyboard shortcuts (like some desktop environments have).

In principle it wouldn't have to be complicated; it could just be a an image that is presented when you press a key combination, e.g. `MOD+XK_question`.

This could be presented using something like sxiv or pqiv, e.g.

```bash
$ pqiv  --transparent-background --hide-info-box /path/to/overlay.png --window-title="Keys" --fade --fullscreen
```

As for the image itself, it could potentially be dynamically generated using a script. Such a script could parse the dwm `config.h` file to extract keybindings and potentially associate certain functions with predefined icons. Potentially this could also parse sxhkd config as well.

There are several graphics libraries for Python that could be useful for placing icons and text on a canvas and outputting the lot as a .png file.

**Comment by yogeshdofficial** [comment]
Hey I'm trying to make a tool for this in python I finished the parsing the config.h and getting the keybindings but I can't find a way to convert them to image for displaying you have mentioned that there are tools for that ,like what?

**Reply by bakkeby** [reply]
I didn't look too far into this but the [pillow](https://pillow.readthedocs.io/en/stable/) library seemed like it could work well. For some functions one might want to have special icons that could be loaded from an image file as an example.

For drawing text there is this example:
https://www.geeksforgeeks.org/python-pillow-writing-text-on-image/
